### PR TITLE
Improve display of 'View determination' action

### PR DESCRIPTION
**Pull request by peteandrew**
This PR makes the 'View determination' action prominent when an application has been determined (accepted or rejected). The button is visible to any user that has access to the submission.

### CODE REVIEW

File: hypha/apply/funds/tests/test_views.py [hunk 1]
```diff
@@ -643,6 +755,7 @@ def test_can_screen_applications_in_final_status(self):
         we can still screen it because we are super user
         """
         submission = ApplicationSubmissionFactory(rejected=True)
+        DeterminationFactory(submission=submission, rejected=True, submitted=True)
```
Comment (todd-dembrey) [inline hypha/apply/funds/tests/test_views.py:758]: Why was this needed here?

File: hypha/apply/determinations/tests/test_views.py [hunk 2]
```diff
@@ -460,7 +461,7 @@ def get_kwargs(self, instance):
 
     def test_can_edit_determination(self):
         submission = ApplicationSubmissionFactory(status='rejected')
-        determination = DeterminationFactory(submission=submission)
+        determination = DeterminationFactory(submission=submission, author=self.user, accepted=True, submitted=True)
```
Comment (todd-dembrey) [inline hypha/apply/determinations/tests/test_views.py:464]: This seems wrong, a submitted determination is not editable i think. Also why did you need to change ownership of the determination, as previously mentioned that isnt important (as long as you are staff).

File: hypha/apply/determinations/tests/test_views.py [hunk 3]
```diff
@@ -60,6 +60,7 @@ def test_can_access_form_if_lead(self):
 
     def test_cant_access_wrong_status(self):
         submission = ApplicationSubmissionFactory(status='rejected')
+        DeterminationFactory(submission=submission, author=self.user, rejected=True, submitted=True)
```
Comment (todd-dembrey) [inline hypha/apply/determinations/tests/test_views.py:63]: Hmm, i think 
https://github.com/OpenTechFund/hypha/blob/ceb74015590ae24d1412478505faeda6951c080d/hypha/apply/determinations/tests/factories.py#L33
should also pass `submission__rejected=True` so that a submission of the right shape is created.

File: hypha/apply/funds/templates/funds/includes/admin_primary_actions.html [hunk 4]
```diff
@@ -14,6 +14,12 @@ <h5>Actions to take</h5>
 
 <a data-fancybox data-src="#screen-application" class="button button--bottom-space button--primary button--full-width {% if screening_form.should_show %}is-not-disabled{% else %}is-disabled{% endif %}" href="#">Screen application</a>
 
+{% if object.is_determined %}
+    {% with determination=object.determinations.final.first %}
```
Comment (todd-dembrey) [inline hypha/apply/funds/templates/funds/includes/admin_primary_actions.html:18]: we want the latest determination here i think so `last`?

File: hypha/apply/funds/templates/funds/includes/generic_primary_actions.html [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{% load primaryactions_tags %}
+
+{% if request.user|should_display_primary_actions_block:object %}
```
Comment (todd-dembrey) [inline hypha/apply/funds/templates/funds/includes/generic_primary_actions.html:3]: This will be fun to maintain & debug. Cant think of something better at this point in time.

File: hypha/apply/funds/templates/funds/includes/admin_primary_actions.html [hunk 6]
```diff
@@ -14,6 +14,12 @@ <h5>Actions to take</h5>
 
 <a data-fancybox data-src="#screen-application" class="button button--bottom-space button--primary button--full-width {% if screening_form.should_show %}is-not-disabled{% else %}is-disabled{% endif %}" href="#">Screen application</a>
 
+{% if object.is_determined %}
```
Comment (todd-dembrey) [inline hypha/apply/funds/templates/funds/includes/admin_primary_actions.html:17]: Whats interesting here is that `is_determined` is coming from the submission state and not from the presence of an actual determination. Which is by design, the determination object is only extra data that doesn't affect the state of the object.

I think what needs to happen is that `object.is_determined` is renamed to something else to reflect the state is final or closed and then we want to be confirming that the determination exists as a separate task. So the submission says that its in a state to show determinations as a primary action and then we check the determinations to confirm they exists.

```
{% if object.is_finished %}
    {% with determination=object.determinations.final.last %}
    {% if determination %}
        ...
    {% endif %}
{% endif %}
```

Comment (peteandrew) [reply hypha/apply/funds/templates/funds/includes/admin_primary_actions.html]: Yeah, this makes more sense. The unecessary test changes were the result of my not understanding that an application being finished doesn't necessarily mean it will have a determination. I've made this change and reverted the previously changed tests

File: hypha/apply/funds/tests/test_views.py (hunk 1 above)
Comment (peteandrew) [reply hypha/apply/funds/tests/test_views.py]: Reverted this after making the change below

File: hypha/apply/determinations/tests/test_views.py (hunk 2 above)
Comment (peteandrew) [reply hypha/apply/determinations/tests/test_views.py]: Reverted this after making the change below

File: hypha/apply/determinations/tests/test_views.py (hunk 3 above)
Comment (peteandrew) [reply hypha/apply/determinations/tests/test_views.py]: Reverted this after making the change below

File: hypha/apply/funds/templates/funds/includes/admin_primary_actions.html (hunk 4 above)
Comment (peteandrew) [reply hypha/apply/funds/templates/funds/includes/admin_primary_actions.html]: I don't think it would make much difference to use first or last as the view guarentees there's only one determination but I've changed this to be consistent with generic_primary_actions.html

File: hypha/apply/funds/templates/funds/includes/generic_primary_actions.html (hunk 5 above)
Comment (peteandrew) [reply hypha/apply/funds/templates/funds/includes/generic_primary_actions.html:3]: Yeah, I know. Not pretty but I can't think of a better solution either
